### ISSUE TITLE
Manage applications in multiple environments on separate Kubernetes clusters

### ISSUE THREAD

**Issue by rynowak** [radius-project/radius#6934]
## Overview of feature request

Radius should be able to manage applications across environments that are not on the same Kubernetes cluster as the control plane.

Examples:

- Radius is installed in Kubernetes cluster `control-plane`. It can manage applications on Kubernetes clusters `prod-westus` and `prod-eu-2` using environments that target those clusters (`env-prod-westus`, `env-prod-eu-2`).
- Radius is running on a VM. It can manage applications on Kubernetes clusters `prod-westus` and `prod-eu-2` using environments that target those clusters (`env-prod-westus`, `env-prod-eu-2`).

**Out of scope:** Defining an environment that spans multiple compute clusters. Each environment will map to a single Kubernetes cluster. A Kubernetes cluster can still contain multiple environments.

## Acceptance criteria

- Decouple Radius from the Kubernetes cluster where applications are running.
- Define environments in a way that identifies and holds credentials for a specific Kubernetes cluster.
- Manage applications on (potentially multiple) Kubernetes clusters. Use the environment to select the destination.

[AB#10707](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10707)

**Comment by sylvainsf** [comment]
## Status update: multi-cluster work is landing in two stages

The design and implementation for deploying to external clusters is underway in #12106 (`multi-cluster` branch), with the technical design in `eng/design-notes/environments/2026-06-multi-cluster.md`.

Heads up that this is being delivered in two stages, and **this issue will stay open until the second stage lands**:

- **v1 (in progress, #12106):** a single `ClusterAccessResolver` seam plus an injected-kubeconfig path (`RADIUS_TARGET_KUBECONFIG` / `KUBE_CONFIG_PATH`). This unblocks deploying recipes to an external EKS/AKS cluster via the Repo Radius `verify`/`deploy` workflows, but the target cluster is supplied as a mounted kubeconfig — **not** named on the environment resource.
- **v2 (planned):** the named-cluster UX this issue actually describes — `providers.aws.eksClusterName` / `providers.azure.aksClusterName` on the environment, with Radius deriving cluster credentials in-process from the registered cloud credential. This is the stage that lets users "manage applications in multiple environments on separate clusters" directly through Radius.

So v1 is a partial step (hence #12106 is "Part of" rather than "Fixes" this issue). I'll close #6934 when v2 ships the named-cluster experience.

**Comment by nicolejms** [comment]
Current implementation and remaining work (reviewed against main `ec979015e67de6222b2b83c462de1ae70790c3e4`).

### Implemented
- Multi-cluster v1 is merged: `ClusterAccessResolver` supports an injected target kubeconfig as well as the local cluster.
- The injected-kubeconfig path enables external-cluster recipe deployment.

### Remaining gaps
- Deliver the per-environment named-cluster experience described for v2, including AKS/EKS selection and credential acquisition from registered cloud credentials.
- The target is currently supplied externally, not selected independently by each environment. The maintainer's two-stage update explicitly leaves this issue open for v2.

References: [Merged v1 implementation (#12106)](https://github.com/radius-project/radius/pull/12106), [two-stage scope clarification](https://github.com/radius-project/radius/issues/6934#issuecomment-4711858430), [resolver strategies](https://github.com/radius-project/radius/blob/ec979015e67de6222b2b83c462de1ae70790c3e4/pkg/recipes/kubernetes/clusteraccess/resolver.go#L91-L100).
